setup.py

```diff
@@ -64,7 +64,20 @@
         "teachable": ["chromadb"],
         "lmm": ["replicate", "pillow"],
         "graph": ["networkx", "matplotlib"],
-        "websurfer": ["beautifulsoup4", "markdownify", "pdfminer.six", "pathvalidate"],
+        "websurfer": [
+            "beautifulsoup4",
```

Comment:
It would be good to list minimum and maximum versions. E.g.
```python
"beautifulsoup4>=4.12,<5"
```
New major version surely break everything and it will happen for some packages sooner or later.